app/invocation/invocation_file_card.tsx

```diff
@@ -123,6 +129,36 @@ export default class InvocationExecLogCardComponent extends React.Component<Prop
     }
   }
 
+  sortDirectories(a: tools.protos.ExecLogEntry, b: tools.protos.ExecLogEntry): number {
+    let first = this.state.direction == "asc" ? a : b;
+    let second = this.state.direction == "asc" ? b : a;
+
+    switch (this.state.sort) {
+      case "size":
+        return +this.sumFileSizes(first?.directory?.files) - +this.sumFileSizes(second?.directory?.files);
+      default:
+        return (first.directory?.path || "").localeCompare(second.directory?.path || "");
+    }
+  }
+
+  sumFileSizes(files: Array<tools.protos.ExecLogEntry.File> | undefined) {
+    return (files || []).map((f) => f.digest?.sizeBytes || 0).reduce((a, b) => new Long(+a + +b)) || Long.ZERO;
+  }
+
+  hashFileHashes(files: Array<tools.protos.ExecLogEntry.File> | undefined) {
+    // Just do something simple here that gives us a unique string that changes when one file changes.
+    return (files || [])
+      .map((f) => f.digest?.hash || "")
+      .reduce((a, b) => {
+        var hash = "";
+        for (var i = 0; i < a.length; i++) {
+          let char = (a.charCodeAt(i) + b.charCodeAt(i)) % 36;
+          hash += String.fromCharCode(char < 26 ? char + 97 : char + 22);
+        }
+        return hash;
+      });
```

Comment:
need an initial value for the case where the list is empty:

```suggestion
      }, "");
```

Reply:
Done